beefy-gadget/src/worker.rs

```diff
@@ -313,6 +317,14 @@ where
 
 				debug!(target: "beefy", "🥩 Round #{} concluded, committed: {:?}.", round.1, signed_commitment);
 
+				if self
+					.backend
+					.append_justification(BlockId::Number(round.1), (BEEFY_ENGINE_ID, signed_commitment.encode()))
+					.is_err()
+				{
+					error!(target: "beefy", "🥩 Failed to append justification: {:?}", signed_commitment);
```

Comment:
Since this is happening inside a `handle_vote`, what are the guarantees that it is never appended twice? I.e. how do we control the rounds currently?

Reply:
Trying to append it a second time will lead to a [Duplicate consensus engine ID](https://github.com/paritytech/substrate/blob/3f110196163b5ec03bac5ee188d60bedf3ebd91d/client/db/src/lib.rs#L1815) error.